src/navigation/pathcurve.cpp

```diff
@@ -84,49 +95,89 @@ void PathCurve::initializeParameterData() {
     }
     _totalLength = _lengthSums.back();
 
+    if (_totalLength < LengthEpsilon) {
+        throw TooShortPathError("Path too short!");
+    }
+
     // Compute a map of arc lengths s and curve parameters u, for reparameterization
-    _parameterSamples.reserve(NrSamplesPerSegment * _nSegments + 1);
-    const double uStep = 1.0 / (_nSegments * NrSamplesPerSegment);
+    constexpr const int steps = 100;
+    const double uStep = 1.0 / static_cast<double>(steps);
+    _parameterSamples.reserve(steps * _nSegments + 1);
+
+    bool problematic = false;
+
     for (unsigned int i = 0; i < _nSegments; i++) {
         double uStart = _curveParameterSteps[i];
         double sStart = _lengthSums[i];
-        for (int j = 0; j < NrSamplesPerSegment; ++j) {
+        _parameterSamples.push_back({ uStart, sStart });
+        // Intermediate sampels
+        for (int j = 1; j < steps; ++j) {
             double u = uStart + j * uStep;
             double s = sStart + arcLength(uStart, u);
+            // Ignore samples that are indistinguishable due to precision limitations
+            if (std::abs(s - _parameterSamples.back().s) < LengthEpsilon) {
+                problematic = true;
+                continue;
+            }
             _parameterSamples.push_back({ u, s });
         }
     }
-    _parameterSamples.push_back({ 1.0, _totalLength });
+
+    if (problematic) {
+        throw InsufficientPrecisionError("Insufficient precision due to path length");
+    }
+
+    // Remove the very last sample if indistinguishable from the final one
+    const double diff = std::abs(_totalLength - _parameterSamples.back().s);
+    if (diff < LengthEpsilon) {
+        _parameterSamples.pop_back();
+    }
+
+    _parameterSamples.push_back({ max, _totalLength });
+    _parameterSamples.shrink_to_fit();
 }
 
 // Compute the curve parameter from an arc length value, using a combination of
 // Newton's method and bisection. Source:
 // https://www.geometrictools.com/Documentation/MovingAlongCurveSpecifiedSpeed.pdf
 // Input s is a length value, in the range [0, _totalLength]
-// Returns curve parameter in range [0, 1]
+// Returns curve parameter in range [0, _nSegments]
 double PathCurve::curveParameter(double s) const {
     if (s <= 0.0) return 0.0;
```

Comment:
Return statements for these two lines should be on the next line with `{` `}`  but I'm open for discussing changing this at one of the developer meetings